File: mobile/src/types/medical.ts
export interface Hospital {
  id: string;
  user_id: string;
  name: string;
  address: string | null;
  phone: string | null;
  latitude: number | null;
  longitude: number | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface HospitalCreate {
  name: string;
  address?: string | null;
  phone?: string | null;
  latitude?: number | null;
  longitude?: number | null;
  notes?: string | null;
}

export interface Doctor {
  id: string;
  user_id: string;
  hospital_id: string | null;
  name: string;
  qualification: string | null;
  specialization: string | null;
  phone: string | null;
  registration_id: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface DoctorCreate {
  hospital_id?: string | null;
  name: string;
  specialization?: string | null;
  phone?: string | null;
  registration_id?: string | null;
  notes?: string | null;
}

export interface Medicine {
  id: string;
  prescription_id: string;
  name: string;
  dosage: string | null;
  frequency: string | null;
  duration_days: number | null;
  timing: string | null;
  morning: boolean;
  afternoon: boolean;
  night: boolean;
  start_date: string | null;
  end_date: string | null;
  is_active: boolean;
  created_at: string;
}

export interface MedicineCreate {
  name: string;
  dosage?: string | null;
  frequency?: string | null;
  duration_days?: number | null;
  timing?: string | null;
  morning?: boolean;
  afternoon?: boolean;
  night?: boolean;
  start_date?: string | null;
  end_date?: string | null;
  is_active?: boolean;
}

export interface Prescription {
  id: string;
  user_id: string;
  family_member_id: string | null;
  doctor_id: string | null;
  hospital_id: string | null;
  prescription_date: string;
  diagnosis: string | null;
  reason_for_visit: string | null;
  image_url: string | null;
  follow_up_date: string | null;
  notes: string | null;
  medicines: Medicine[];
  created_at: string;
  updated_at: string;
}

export interface PrescriptionCreate {
  family_member_id?: string | null;
  doctor_id?: string | null;
  hospital_id?: string | null;
  prescription_date: string;
  diagnosis?: string | null;
  reason_for_visit?: string | null;
  image_url?: string | null;
  follow_up_date?: string | null;
  notes?: string | null;
  medicines: MedicineCreate[];
}

export interface ActiveMedicine {
  id: string;
  prescription_id: string;
  name: string;
  dosage: string | null;
  frequency: string | null;
  timing: string | null;
  morning: boolean;
  afternoon: boolean;
  night: boolean;
  start_date: string | null;
  end_date: string | null;
  doctor_name: string | null;
  diagnosis: string | null;
}

export interface HospitalDetail extends Hospital {
  last_visit_date: string | null;
  visit_count: number;
  doctors: Doctor[];
}

export interface VisitSummary {
  prescription_id: string;
  hospital_id: string | null;
  hospital_name: string | null;
  hospital_address: string | null;
  hospital_phone: string | null;
  visit_date: string;
  patient_name: string | null;
  reason_for_visit: string | null;
  diagnosis: string | null;
  doctor_name: string | null;
  doctor_qualification: string | null;
  doctor_registration_id: string | null;
  medicines: Medicine[];
}

export interface DoctorDetail extends Doctor {
  last_visit_date: string | null;
  visit_count: number;
}

// OCR extraction result (before saving)
export interface OCRMedicine {
  name: string;
  dosage: string | null;
  frequency: string | null;
  duration: string | null;
  timing: string | null;
  morning: boolean;
  afternoon: boolean;
  night: boolean;
}

export interface OCRPrescriptionResult {
  doctor_name: string | null;
  doctor_qualification: string | null;
  doctor_registration_id: string | null;
  hospital_name: string | null;
  hospital_address: string | null;
  hospital_phone: string | null;
  visit_date: string | null;
  diagnosis: string | null;
  reason_for_visit: string | null;
  medicines: OCRMedicine[];
  follow_up_date: string | null;
  raw_text: string | null;
}

export const SPECIALIZATIONS = [
  'General Physician', 'Cardiologist', 'Dermatologist', 'ENT',
  'Gynecologist', 'Neurologist', 'Ophthalmologist', 'Orthopedic',
  'Pediatrician', 'Psychiatrist', 'Pulmonologist', 'Urologist', 'Other',
] as const;

export const TIMINGS = [
  { value: 'before_food', label: 'Before Food' },
  { value: 'after_food', label: 'After Food' },
  { value: 'with_food', label: 'With Food' },
] as const;

File: backend/app/schemas/medical.py
from pydantic import BaseModel
from uuid import UUID
from datetime import date, datetime
from typing import Optional
from decimal import Decimal


# --- Hospital ---
class HospitalCreate(BaseModel):
    name: str
    address: Optional[str] = None
    phone: Optional[str] = None
    latitude: Optional[Decimal] = None
    longitude: Optional[Decimal] = None
    notes: Optional[str] = None


class HospitalUpdate(BaseModel):
    name: Optional[str] = None
    address: Optional[str] = None
    phone: Optional[str] = None
    latitude: Optional[Decimal] = None
    longitude: Optional[Decimal] = None
    notes: Optional[str] = None


class HospitalResponse(BaseModel):
    id: UUID
    user_id: UUID
    name: str
    address: Optional[str] = None
    phone: Optional[str] = None
    latitude: Optional[Decimal] = None
    longitude: Optional[Decimal] = None
    notes: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


# --- Doctor ---
class DoctorCreate(BaseModel):
    hospital_id: Optional[UUID] = None
    name: str
    qualification: Optional[str] = None
    specialization: Optional[str] = None
    phone: Optional[str] = None
    registration_id: Optional[str] = None
    notes: Optional[str] = None


class DoctorUpdate(BaseModel):
    hospital_id: Optional[UUID] = None
    name: Optional[str] = None
    qualification: Optional[str] = None
    specialization: Optional[str] = None
    phone: Optional[str] = None
    registration_id: Optional[str] = None
    notes: Optional[str] = None


class DoctorResponse(BaseModel):
    id: UUID
    user_id: UUID
    hospital_id: Optional[UUID] = None
    name: str
    qualification: Optional[str] = None
    specialization: Optional[str] = None
    phone: Optional[str] = None
    registration_id: Optional[str] = None
    notes: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


# --- Medicine ---
class MedicineCreate(BaseModel):
    name: str
    dosage: Optional[str] = None
    frequency: Optional[str] = None
    duration_days: Optional[int] = None
    timing: Optional[str] = None
    morning: bool = False
    afternoon: bool = False
    night: bool = False
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    is_active: bool = True


class MedicineResponse(BaseModel):
    id: UUID
    prescription_id: UUID
    name: str
    dosage: Optional[str] = None
    frequency: Optional[str] = None
    duration_days: Optional[int] = None
    timing: Optional[str] = None
    morning: bool
    afternoon: bool
    night: bool
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    is_active: bool
    created_at: datetime

    model_config = {"from_attributes": True}


# --- Prescription ---
class PrescriptionCreate(BaseModel):
    family_member_id: Optional[UUID] = None
    doctor_id: Optional[UUID] = None
    hospital_id: Optional[UUID] = None
    prescription_date: date
    diagnosis: Optional[str] = None
    reason_for_visit: Optional[str] = None
    image_url: Optional[str] = None
    follow_up_date: Optional[date] = None
    notes: Optional[str] = None
    medicines: list[MedicineCreate] = []


class PrescriptionUpdate(BaseModel):
    family_member_id: Optional[UUID] = None
    doctor_id: Optional[UUID] = None
    hospital_id: Optional[UUID] = None
    prescription_date: Optional[date] = None
    diagnosis: Optional[str] = None
    reason_for_visit: Optional[str] = None
    image_url: Optional[str] = None
    follow_up_date: Optional[date] = None
    notes: Optional[str] = None


class PrescriptionResponse(BaseModel):
    id: UUID
    user_id: UUID
    family_member_id: Optional[UUID] = None
    doctor_id: Optional[UUID] = None
    hospital_id: Optional[UUID] = None
    prescription_date: date
    diagnosis: Optional[str] = None
    reason_for_visit: Optional[str] = None
    image_url: Optional[str] = None
    follow_up_date: Optional[date] = None
    notes: Optional[str] = None
    medicines: list[MedicineResponse] = []
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


# --- Detail Responses ---
class HospitalDetailResponse(HospitalResponse):
    last_visit_date: Optional[date] = None
    visit_count: int = 0
    doctors: list[DoctorResponse] = []


class DoctorDetailResponse(DoctorResponse):
    last_visit_date: Optional[date] = None
    visit_count: int = 0


# --- Visit Summary (for the Medical tab list) ---
class VisitSummary(BaseModel):
    prescription_id: UUID
    hospital_id: Optional[UUID] = None
    hospital_name: Optional[str] = None
    hospital_address: Optional[str] = None
    hospital_phone: Optional[str] = None
    visit_date: date
    patient_name: Optional[str] = None
    reason_for_visit: Optional[str] = None
    diagnosis: Optional[str] = None
    doctor_name: Optional[str] = None
    doctor_qualification: Optional[str] = None
    doctor_registration_id: Optional[str] = None
    medicines: list["MedicineResponse"] = []


# --- Active Medicines ---
class ActiveMedicineResponse(BaseModel):
    id: UUID
    prescription_id: UUID
    name: str
    dosage: Optional[str] = None
    frequency: Optional[str] = None
    timing: Optional[str] = None
    morning: bool
    afternoon: bool
    night: bool
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    doctor_name: Optional[str] = None
    diagnosis: Optional[str] = None
